### PR TITLE
fix: Fix emails related issues

### PR DESCRIPTION
**Pull request by soniaklimas**
I want to merge this change because it fixes the emails related issue:
* user being redirected from email change confirmation link was able to directly change url from /sign-in to /account/profile and see empty user profile page;
* user after refreshing old session before email change was not redirected to /sign-in page;
* user after clicking again on email change confirmation link saw wrong error message.

# Pull Request Checklist

- [ ] Test the changes locally to ensure they work as expected.
- [ ] Document the testing process and results in the pull request description. (Screen recording, screenshot etc)
- [ ] Include new tests for any new functionality or significant changes.
- [ ] Ensure that tests cover edge cases and potential failure points.
- [ ] Document the impact of the changes on the system, including potential risks and benefits.
- [ ] Provide a rollback plan in case the changes introduce critical issues.
- [ ] Update documentation to reflect any changes in functionality.

### CODE REVIEW

File: apps/storefront/src/app/[locale]/(auth)/confirm-new-email/page.tsx [hunk 1]
```diff
@@ -9,39 +9,25 @@ import { userService } from "@/services";
...
+    accessToken,
+    channel: region.market.channel,
+    token,
+  });
 
-      return t("auth.too-much-time-has-passed");
-    }
+  if (result.ok) {
+    redirect({ href: "/api/logout", locale });
   }
```
Comment (GrzegorzDerdak) [inline apps/storefront/src/app/[locale]/(auth)/confirm-new-email/page.tsx:32]: Maybe we could create a server action called `confirmEmailchangeAction` that will encapsulate this logic:
- call the `userService.confirmEmailChange`
- if success, remove cookies and redirect to /sign-in`
- if failed, return an error?

Can you check if its doable?

Also there's already an API route for deleting cookies - it accepts one cookieKey, but maybe can be extended to accept multiple keys

File: packages/infrastructure/src/user/saleor/infrastructure/confirm-email-change-infra.ts [hunk 2]
```diff
@@ -32,11 +33,7 @@ export const saleorConfirmEmailChangeInfra =
         error: result.data.confirmEmailChange.errors,
       });
 
-      return err([
-        {
-          code: "EMAIL_CHANGE_CONFIRMATION_ERROR",
-        },
-      ]);
+      handleMutationErrors(result.data?.confirmEmailChange?.errors);
```
Comment (GrzegorzDerdak) [inline packages/infrastructure/src/user/saleor/infrastructure/confirm-email-change-infra.ts:36]:
```suggestion
      return err(handleMutationErrors(result.data.confirmEmailChange.errors));

```

File: apps/storefront/messages/en-GB.json [hunk 3]
```diff
@@ -269,6 +269,8 @@
     "INPUT_ERROR": "Input error.",
     "INSUFFICIENT_STOCK_ERROR": "Insufficient stock.",
     "INVALID_VALUE_ERROR": "Invalid value.",
+    "JWT_INVALID_TOKEN_ERROR": "Invalid token value",
```
Comment (GrzegorzDerdak) [inline apps/storefront/messages/en-GB.json:272]: I'd provide a better message - the customer does not know nothing about tokens etc. Something more descriptive :)

**Review by GrzegorzDerdak — CHANGES_REQUESTED** [review]

File: apps/storefront/src/app/[locale]/(auth)/confirm-new-email/page.tsx (hunk 1 above)
Comment (soniaklimas) [reply apps/storefront/src/app/[locale]/(auth)/confirm-new-email/page.tsx:32]: I created server action as you suggested but with note - because in a page, fetch runs within the user's request-response lifecycle and can modify cookies via API routes, but in a server action, fetch runs in isolation and its Set-Cookie headers are not applied to the user's response - that's why we cannot use already existed cookies/delete route handler.

File: apps/storefront/messages/en-GB.json (hunk 3 above)
Comment (soniaklimas) [reply apps/storefront/messages/en-GB.json:272]: The most difficult part of this task :D

**Review by GrzegorzDerdak — APPROVED** [review]
